source/grid/grid_generator.cc

```diff
@@ -4298,175 +4298,60 @@ namespace GridGenerator
     const double                  duplicated_vertex_tolerance,
     const bool                    copy_manifold_ids)
   {
+    std::vector<Point<spacedim>> vertices;
+    std::vector<CellData<dim>>   cells;
+    SubCellData                  subcell_data;
+
+    unsigned int n_accumulated_vertices = 0;
     for (const auto triangulation : triangulations)
       {
-        (void)triangulation;
         Assert(triangulation->n_levels() == 1,
                ExcMessage("The input triangulations must be non-empty "
                           "and must not be refined."));
-      }
-
-    // get the union of the set of vertices
-    unsigned int n_vertices = 0;
-    for (const auto triangulation : triangulations)
-      n_vertices += triangulation->n_vertices();
 
-    std::vector<Point<spacedim>> vertices;
-    vertices.reserve(n_vertices);
-    for (const auto triangulation : triangulations)
-      vertices.insert(vertices.end(),
-                      triangulation->get_vertices().begin(),
-                      triangulation->get_vertices().end());
-
-    // now form the union of the set of cells
-    std::vector<CellData<dim>> cells;
-    unsigned int               n_cells = 0;
-    for (const auto triangulation : triangulations)
-      n_cells += triangulation->n_cells();
+        std::vector<Point<spacedim>> tria_vertices;
+        std::vector<CellData<dim>>   tria_cells;
+        SubCellData                  tria_subcell_data;
+        std::tie(tria_vertices, tria_cells, tria_subcell_data) =
```

Comment:
Nice -- though wouldn't you wish we had structured bindings already? https://en.cppreference.com/w/cpp/language/structured_binding

Reply:
I must admit that I still like typing out the types, though I could get rid of three lines of code. I'm not sure which I like more :)